### PR TITLE
[WIP] Socket activated systemd

### PR DESCRIPTION
**Pull request by Jackbaude**
socket activated service for the nv-dhcp server

Signed-off-by: Jack <jackbaude@gmail.com>

### CODE REVIEW

File: contrib/systemd/nv-dhcp.service [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[Unit]
+Description=Netavark DHCP proxy
+Requires=nv-dhcp.socket
+After=nv-dhcp.socket
+StartLimitIntervalSec=0
+
+[Service]
+Type=exec
+Delegate=true
```
Comment (vrothberg) [inline contrib/systemd/nv-dhcp.service:9]: What's the reasoning for setting it on?  Is the proxy fiddling with cgroups?

File: contrib/systemd/nv-dhcp.service [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+Restart=always
+KillMode=process
```
Comment (vrothberg) [inline contrib/systemd/nv-dhcp.service:11]: `KillMode=process` is being discouraged by systemd and will yield warnings. I recommend leaving the default which is "control-group" which will nuke everything in the cgroup of the service.  That should be enough even when the proxy forks or execs other processes.

**Comment by Jackbaude** [comment]
hey @vrothberg I saw your requests for changes and I will get to them. What is more concerning is that the socket activation does not seem to be working. To reproduce:

1. Start the service
2. Request a lease
3. Stop the service
4. Request a lease 

However, when the service is stopped, the service should come back up by the socket activation but it does not. Is there something that I am missing in the `.service` and `.socket` files, or does this have to due with the server itself? I dont have much understanding about how these systems work so please feel free to ask clarifying questions. Thank you

File: contrib/systemd/nv-dhcp.service [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[Unit]
+Description=Netavark DHCP proxy
+Requires=nv-dhcp.socket
+After=nv-dhcp.socket
+StartLimitIntervalSec=0
+
+[Service]
+Type=exec
+Delegate=true
+Restart=always
```
Comment (vrothberg) [inline contrib/systemd/nv-dhcp.service:10]: Maybe `always` is causing issue?

File: contrib/systemd/nv-dhcp.socket [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[Unit]
+Description=Netavark DHCP Proxy for per-connection netavark clients
+
+[Socket]
+ListenStream=%t/nv-dhcp/nv-dhcp-proxy.sock
+Accept=yes
```
Comment (vrothberg) [inline contrib/systemd/nv-dhcp.socket:6]: `Accept=yes` is probably not needed. As the name of .socket and .service matches.

**Review by vrothberg — COMMENTED** [review]
@Jackbaude can you try without `Restart=always`?  Please let me know if that resolves your issue.

**Comment by baude** [comment]
consider something like the following:

```diff --git a/contrib/systemd/nv-dhcp.service b/contrib/systemd/nv-dhcp.service
index f577574..66689cc 100644
--- a/contrib/systemd/nv-dhcp.service
+++ b/contrib/systemd/nv-dhcp.service
@@ -9,7 +9,7 @@ Type=exec
 Delegate=true
 Restart=always
 KillMode=process
-ExecStart=/usr/local/bin/netavark-proxy
+ExecStart=/usr/libexec/podman/netavark-proxy
 
 [Install]
 WantedBy=default.target
diff --git a/contrib/systemd/nv-dhcp.socket b/contrib/systemd/nv-dhcp.socket
index 7d17033..6e660aa 100644
--- a/contrib/systemd/nv-dhcp.socket
+++ b/contrib/systemd/nv-dhcp.socket
@@ -2,5 +2,4 @@
 Description=Netavark DHCP Proxy for per-connection netavark clients
 
 [Socket]
-ListenStream=%t/nv-dhcp/nv-dhcp-proxy.sock
-Accept=yes
+ListenStream=%t/podman/nv-proxy.sock
diff --git a/src/lib.rs b/src/lib.rs
index 113eeb2..ae51a94 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -8,7 +8,7 @@ pub mod dhcp_service;
 use std::fs::File;
 // TODO these constant destinations are not final.
 // Default UDS path for gRPC to communicate on.
-pub const DEFAULT_UDS_PATH: &str = "/run/nv-dhcp/nv-dhcp-uds.sock";
+pub const DEFAULT_UDS_PATH: &str = "/run/podman/nv-proxy.sock";
 // Default configuration directory.
 pub const DEFAULT_CONFIG_DIR: &str = "";
 // Default Network configuration path
```

**Comment by baude** [comment]
also lets settle on a name, i like netavark-proxy best ... so lets name everything like that

**Comment by baude** [comment]
also remove the Accepts as @vrothberg advised

**Pull request closed without merging**